python/cudf/cudf/core/series.py

```diff
@@ -453,31 +452,20 @@ def empty(self):
         return not len(self)
 
     def __getitem__(self, arg):
-        data = self._column[arg]
-        index = self.index.take(arg)
-        if is_scalar(data) or data is None:
-            return data
-        return self._copy_construct(data=data, index=index)
-
-    def __setitem__(self, key, value):
-        # coerce value into a scalar or column
-        if is_scalar(value):
-            value = to_cudf_compatible_scalar(value)
+        if isinstance(arg, (slice, range, ColumnBase)) or (
+            pd.api.types.is_bool_dtype(as_column(arg).dtype)
```

Comment:
Converting this to a column to get the dtype is crazy expensive. Couldn't we just always pass this to `.loc` which also needs to handle boolean masking anyway? Slicing should slice the index anyway, right?

Reply:
loc doesn't handle boolean mask, we may have to introduce this to loc.

Reply:
I am sorry for wrong information, `cudf.loc` supports boolean masking.